pyracf/common/irrsmo00.py

```diff
@@ -19,9 +20,29 @@ def __init__(self) -> None:
         # Initialize size of output buffer
         self.buffer_size = 100000
 
-    def call_racf(self, request_xml: bytes, precheck: bool = False) -> str:
+    def call_racf(
```

Comment:
Suggested refactoring. This should be functionally the same, but just a little more concise and cleaner.

```
    def call_racf(
        self,
        request_xml: bytes,
        precheck: bool = False,
        run_as_userid: Union[str, None] = None,
    ) -> str:
        """Make request to call_irrsmo00 in the cpyracf Python extension."""
        options = 15 if precheck else 13
        userid = b""
        userid_length = 0
        if run_as_userid:
            userid = run_as_userid.encode("cp1047")
            userid_length = len(run_as_userid)
        response = call_irrsmo00(
            xml_str=request_xml,
            xml_len=len(request_xml),
            opts=options,
            userid=userid,
            userid_len=userid_length
        ).decode("cp1047")
        if response[0] == "":
            return response[1:3]
        return response[0]
```